web/common/src/components/VirtualList/FilterableList.tsx

```diff
@@ -0,0 +1,112 @@
+import Fuse, { type IFuseOptions } from 'fuse.js'
+import React from 'react'
+
+import { Input } from '../extra/input'
+import { VerticalContainer } from '../VerticalContainer/VerticalContainer'
+import { HorizontalContainer } from '../HorizontalContainer/HorizontalContainer'
+import { Badge } from '../Badge/Badge'
+import { cn } from '@/utils'
+import MessageContainer from '../MessageContainer/MessageContainer'
+
+export function FilterableList<
+  TItem extends Record<string, any> = Record<string, any>,
+>({
+  items,
+  disabled,
+  placeholder,
+  autoFocus,
+  filterOptions,
+  className,
+  children,
+  beforeInput,
+  afterInput,
+}: {
+  items: TItem[]
+  filterOptions?: IFuseOptions<any>
```

Comment:
Injecting the whole search function feels to me like a really simple way to avoid this.